server/src/main/java/io/crate/expression/scalar/ConcatFunction.java

```diff
@@ -41,6 +41,7 @@
 public abstract class ConcatFunction extends Scalar<String, String> {
 
     public static final String NAME = "concat";
+    public static final String OPERATOR_NAME = "_concat_op";
```

Comment:
Could use `Operator.PREFIX + _concat` here for consistency? Although it shouldn't really matter if we keep it internal & hidden

Reply:
Looking at `Operator.PREFIX` usages and https://github.com/crate/crate/blob/15cfe452bba93bbcb2dfa10c6ac9e37627c78729/server/src/main/java/io/crate/expression/symbol/Function.java#L467-L481 I think going with `Operator.PREFIX + "||"` might make the formatting work out of the box but can do that in a follow up.

Reply:
As I forgot to register the operator name also for arrays and objects, I had to push a fixup anyhow. See https://github.com/crate/crate/pull/16295/commits/4d7f7ced3bebbb035a4bd71f5d678bb950cdad60

Reply:
I also added a formatting test. As you said, it works out-of-the-box when using the operator prefix.